caffe2/operators/slice_op.h

```diff
@@ -44,14 +117,28 @@ bool SliceImpl(
   CAFFE_ENFORCE_GE(data.ndim(), starts.size());
   CAFFE_ENFORCE_EQ(starts.size(), ends.size());
 
+  // Compute start and end indices
+  const int last_nonfull_dimension = _last_nonfull_dimension<SIndex>(data.dims(), starts, ends);
```

Comment:
you probably want to collapse several "full" dimensions in a row - that would be more efficient

also, even though it's correct - move it after the disambiguation of dimensions below? that way it's more understandable.